### ISSUE TITLE
Review Map Browse/Map views in general

### ISSUE THREAD

**Issue by taylor-steve** [MukurtuCMS/Mukurtu-CMS#185]
# History
Years ago, we were using a customized leaflet library that provided clustering support for leaflet views. We have a few computed fields `field_mukurtu_geojson`, `field_mukurtu_cluster_lat`, `field_mukurtu_cluster_long` that were in support of that initial effort. That library became unsupported, and we temporarily went back to the standard Drupal leaflet library and mostly disabled the use of those computed fields (but they are still present).

# Recommendation
If clustering is still desired, I'd recommend first checking if it has been added into the main contrib module for Leaflet or if somebody has picked up the torch on a different contrib module. If you find something that works, it'd be best to remove those computed fields and related functions, found primarily in `mukurtu_browse`. 

If neither of those paths work out, it may be necessary to write the extension specifically for Mukurtu, in which case maybe the fragments of the old effort will be useful.

**Comment by michael-wynne-wsu** [comment]
It *does* appear to be in leaflet now, if I'm reading their release information properly: https://www.drupal.org/project/leaflet

**Comment by steve-taylor-wsu** [comment]
Yes, I probably should have dug up more specific information. The kind of clustering we want to do is provided by the separate, but officially supported, plugin `Leaflet.markercluster` which is the one that stopped working for us. We last used version `1.4.1` and they are up to `1.5.3`, so hopefully they've addressed our issues in that change from 1.4 to 1.5. They haven't issued any new releases in over two years. Fingers crossed that represents stability and not that it has become unsupported.

**Comment by quicksketch** [comment]
The presence of these computed fields is complicating the export of Features (see #672), so I'm going to work on removing these fields which are present on _every_ content type.

**Comment by quicksketch** [comment]
This also indicates promising development in the Leaflet module, that it integrated `Leaflet.markercluster` 1.5.3 into the main module about 5 months ago: https://www.drupal.org/project/leaflet/issues/3464444

**Comment by quicksketch** [comment]
PR at https://github.com/MukurtuCMS/Mukurtu-CMS/pull/675 that removes the computed fields is ready for review.

I left in place the Views handler for providing a bounding-box filter, it had only a very loose connection to the `field_mukurtu_geojson` field. However, I am not sure what it does or how to test its functionality. It would be great to have someone a bit more familiar with how mapping works in Mukurtu verify if that views handler is still needed.

**Comment by michael-wynne-wsu** [comment]
@nick-deer @jonscottwsu @quicksketch I think this issue can be closed with the recent Leaflet upgrades. I'm not missing anything here am I?

**Comment by nick-deer** [comment]
@michael-wynne-wsu I think we can close this issue specifically. However, I tried to get clustering to work with the latest Leaflet upgrades but I couldn't get it to show up. Unless I'm missing something, I propose we make a new issue for re-implementing clustering with the updated Leaflet markercluster library.

**Comment by michael-wynne-wsu** [comment]
@nick-deer I'm seeing a little other oddness as well, so I'll document the makercluster and other things and close this.

**Comment by michael-wynne-wsu** [comment]
See https://github.com/MukurtuCMS/Mukurtu-CMS/issues/721
